core/src/test/java/org/elasticsearch/cluster/service/ClusterServiceTests.java

```diff
@@ -370,10 +377,29 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
         }
     }
 
+    public void testSingleBatchSubmission() {
+        Map<Integer, ClusterStateTaskListener> tasks = new HashMap<>();
+        for (int i = randomInt(10); i > 0; i--) {
+            tasks.put(randomInt(1024), (source, t) -> fail(ExceptionsHelper.detailedMessage(t)));
+        }
+
+        clusterService.submitStateUpdateTasks("test", tasks, ClusterStateTaskConfig.build(Priority.LANGUID),
```

Comment:
I think that you need to latch the main test thread to ensure that these execute before the main thread exits the test method?